pkg/cache/v3/resource.go

```diff
@@ -97,6 +97,23 @@ func GetResourceName(res types.Resource) string {
 	}
 }
 
+// ResourceRequiresFullStateInSotw indicates whether when building the reply in Sotw,
+// the response must include all existing resources or can return only the modified ones
+func ResourceRequiresFullStateInSotw(typeURL resource.Type) bool {
+	// From https://www.envoyproxy.io/docs/envoy/v1.28.0/api-docs/xds_protocol#grouping-resources-into-responses,
+	// when using sotw the control-plane MUST return all requested resources (or simply all if wildcard)
+	// for some types. This is relied on by xds-grpc which is explicitly requesting clusters but expect
+	// to receive all existing resources
```

Comment:
In a way, yes, this is a specificity of xDS gRPC, because xDS gRPC uses resource names with LDS and CDS, whereas IIUC, envoy does not. For cases where Envoy does (ODCDS?) then I don't know how resource deletions are handled. I would add that for gRPC clients, it not including a resource in LDS or CDS indicates that the resource has been deleted. See https://github.com/grpc/proposal/blob/master/A53-xds-ignore-resource-deletion.md#background which has more details.